float32 x_coord
float32 y_coord
float32 z_coord
int32 id
int32 action